chia/wallet/wallet_state_manager.py

```diff
@@ -2214,20 +2213,43 @@ async def coin_added(
 
         await self.create_more_puzzle_hashes()
 
-    async def add_pending_transaction(self, tx_record: TransactionRecord) -> None:
+    async def add_pending_transactions(
+        self, tx_records: List[TransactionRecord], merge_spends: bool = True
+    ) -> List[TransactionRecord]:
         """
-        Called from wallet before new transaction is sent to the full_node
+        Add a list of transactions to be submitted to the full node.
+        Aggregates the `spend_bundle` property for each transaction onto the first transaction in the list.
         """
-        # Wallet node will use this queue to retry sending this transaction until full nodes receives it
-        await self.tx_store.add_transaction_record(tx_record)
+        agg_spend: SpendBundle = SpendBundle.aggregate(
+            [tx.spend_bundle for tx in tx_records if tx.spend_bundle is not None]
+        )
+        actual_spend_involved: bool = agg_spend != SpendBundle([], G2Element())
+        if merge_spends and actual_spend_involved:
+            tx_records = [
+                dataclasses.replace(
+                    tx,
+                    spend_bundle=agg_spend if i == 0 else None,
+                    name=agg_spend.name() if i == 0 else bytes32.secret(),
```

Comment:
`bytes32.secret()` is not deterministic. I take it that's not a problem, right?

Reply:
No, as far as I know we have no commitment to the uniformity of transaction IDs so the `name` property is mostly used as a de-dup and random 32 bytes works great for that.

Reply:
Worth noting that I considered just using the existing name here but some wallets set that name property to the spend bundle name (which this function is already doing) and we end up with duplicates.  Rather than have magic knowledge to not accidentally do, I went this route.